.buildkite/scripts/integration_tests.sh

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env bash
+set -exuo pipefail
+
+export WORKSPACE=`pwd`
+
+vault kv get -format=json -field=data kv/ci-shared/observability-ingest/cloud/gcp > ./gcp.json
+export GOOGLE_APPLICATION_CREDENTIALS=$(realpath ./gcp.json)
+export TEST_INTEG_AUTH_GCP_SERVICE_TOKEN_FILE=$(realpath ./gcp.json)  
+
+if ! command -v go &>/dev/null; then  
+  echo "Go is not installed. Installing Go..."
+  export GO_VERSION=`cat .go-version`
+  curl -O https://dl.google.com/go/go$GO_VERSION.linux-amd64.tar.gz
+  sudo tar -xf go$GO_VERSION.linux-amd64.tar.gz -C /usr/local
+  echo 'export PATH=$PATH:/usr/local/go/bin' >> ~/.bashrc
+  source ~/.bashrc
+  mkdir $HOME/go
+  mkdir $HOME/go/bin
+  export PATH=$PATH:/usr/local/go/bin:$HOME/go/bin  
+  echo "Go has been installed."
+else
+  echo "Go is already installed."
+fi
+
+# Installing mage
+make mage
+
+# 
+# PACKAGE
+# 
+
+DEV=true EXTERNAL=true SNAPSHOT=true PLATFORMS=linux/amd64,linux/arm64 PACKAGES=tar.gz mage -v package
+
+#ESS
+vault kv get -field api_key kv/ci-shared/observability-ingest/elastic-agent-ess-qa > ./apiKey
```

Comment:
Same for this secret, would it make sense to move it to pre-command hook ?

Reply:
Moved secrets initialisation to the `pre-command` hook

Reply:
What is the error ? pre-command hook runs in the same agent (same agent in each step of the pipeline)

Reply:
Fixed